---
title: "Exploration of Squirrel"
author: "Alicia Halhed"
date: "14/11/2019"
output: html_document
---
### Attach Required Packages
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
setwd("~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/exploration")

library(tidyverse)
#BiocManager::install("ShortRead")
library(ShortRead) #for reading fasta file
library(seqinr) #for writing fasta file
library(Biostrings)
library(stringr)
library(tibble)
library(data.table)
library(dada2)
library(phyloseq)
library(DECIPHER)
library(qiime2R)
```


### Load in the original data
```{r, OTUs}
# Since the IDs were in a comment line in the text file, I added a new line there (in the text file) to allow R to read them in as column names
rs_otu <- read.table("otu_table_all_samplesAH.txt", sep="\t",
                     header = TRUE, row.names = "ID")
View(rs_otu) # so the taxonomy grouping is in a column at the end of the data and the denovo number is the row name

# Remove the taxonomy column
rs_otu %>% select(-c(taxonomy)) -> rs_otu_tax
View(rs_otu_tax)
```


```{r, Metadata}
rs_metadata <- readxl::read_xlsx("../original/metadata_all_samples.xlsx")
#so they actually typed in NAs, I will replace those with actual NAs
rs_metadata <- rs_metadata %>% mutate_all(funs(str_replace(., "NA", "replace")))
rs_metadata[ rs_metadata == "replace" ] <- NA
View(rs_metadata)
```

```{r}
# add location column for this to upload to the server
rs_metadata %>%
  mutate(Location = paste0(`Location X`, ", ", `Location Y`),
         Date = paste0(Month, Year)) %>%
  write.table("RS_meta.tsv", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
```

#### DO NOT RUN THE FOLLOWING CHUNK IF IT CAN BE AVOIDED
This one is especially slow, but gives us the sequences in a useable format.
```{r, Sequences 2, eval=FALSE}
# read the fasta file into R
rs_fasta <- readDNAStringSet("~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/rs-QIIME2-AH/RS_seqs.fasta")
seq_name <- names(rs_fasta)
sequence <- paste(rs_fasta)
# put the fasta file into a data frame
rs_fasta_df <- data.frame(seq_name, sequence)
# look at the data frame
head(rs_fasta_df)
```

```{r}
# write the full data frame out to a file for QIIME2
# 'SampleData[Sequences]' format in Q2 doesn't seem to like the sequences being "chunked" but would take the continuous (what I did locally)
rs_fasta_df %>%
  mutate(seq_name = paste0(">", seq_name)) %>%
  write.table("RS_seqsR.fasta", sep = "\n", row.names = FALSE, col.names = FALSE, quote = FALSE)
```

### Sampling - Round 1 (General Sampling)
Take a sample of the metadata
```{r}
rs_metadata %>% group_by(Grid) %>% sample_frac(.05) -> rs_sub_metadata
write.table(rs_sub_metadata, "RS_metadata.tsv", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
```

```{r}
rs_sub_metadata2 <- rs_sub_metadata
rs_sub_metadata2$Location <- paste(rs_sub_metadata2$`Location X`, rs_sub_metadata2$`Location Y`, sep = ", ")
rs_sub_metadata2$Date <- paste(rs_sub_metadata2$Month, rs_sub_metadata2$Year, sep = "")
# writing this second version because QIIME was having issues with the squirrel.ID name and needs the locations combined
rs_sub_metadata2 %>% plyr::rename(replace = c("Squirrel.ID" = "SquirrelID")) %>%
  write.table("~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/rs-QIIME2-AH/sampled5/RS_metadata_sub2.tsv", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
```

Sample the OTU table
```{r}
# keep matching column names
rs_sub_otu <- rs_otu[,which(colnames(rs_otu) %in% pull(rs_sub_metadata, SampleID))] %>% t
```
```{r}
#write the new subset of the OTUs to file
rs_sub_ex <- rs_otu[,which(colnames(rs_otu) %in% pull(rs_sub_metadata, SampleID))] #%>% rownames_to_column
rs_sub_ex <- rs_sub_ex[apply(rs_sub_ex[,-1], 1, function(x) !all(x==0)),] %>% rownames_to_column
tax <- rs_otu %>% select(taxonomy) %>% 
  rownames_to_column %>% 
  inner_join(rs_sub_ex, by = "rowname") %>%
  select(-taxonomy, everything()) %>%
  rename("# OTU ID" = rowname) # %>% remove_rownames %>% column_to_rownames(var = "rowname")
write.table(tax, "RS_otu.txt", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
rm(rs_sub_ex)
rm(tax)
```


#### DO NOT RUN THE FOLLOWING CHUNK IF IT CAN BE AVOIDED
Prepare data and write sampled data to FASTA.
```{r, Subset Sequences}
# Extract the SampleID without the rest of the header information (this is what will match to the metadata)
rs_fasta_df$SampleID <- stringr::word(rs_fasta_df$seq_name, 1, sep = "_")
# Match the sequences (all from each host) each sampled indvidual in the metadata
rs_sub_fasta <- inner_join(rs_fasta_df, rs_metadata_sample, by = 'SampleID') %>% select(seq_name, sequence)
#write the sample to fasta
export_file <- rs_sub_fasta %>% select(seq_name, sequence)#only need these two columns (header and sequence) for the fasta file
export_file$seq_name <- paste(">", export_file$seq_name, sep="")
write.table(export_file, "RS_sampled.fasta", sep = "\n", row.names = FALSE, col.names = FALSE, quote = FALSE)
#write.table(export_file, "RS_sampling.fasta", sep = "\n", row.names = FALSE, col.names = FALSE, quote = FALSE)
```

### Preliminary Questions about the data

Do any of the squirrels occur as both parent and sampled individual?
```{r}
rs_metadata_full <- rs_metadata

# Is the sire matched in the SquirrelID column? True = yes, false = no
rs_metadata_full$match.sire <- rs_metadata$SireID %in% rs_metadata$Squirrel.ID
rs_metadata_full$match.dam <- rs_metadata$DamID %in% rs_metadata$Squirrel.ID
View(rs_metadata_full)
```

Are any of the squirrels sampled multiple times? 
```{r}
rs_metadata_multi <- rs_metadata_full %>% 
  group_by(Squirrel.ID) %>% 
  filter(n()>1) %>% 
  arrange(Squirrel.ID)
#so yes, many we sampled multiple times
# return the unique values of Squirrel.ID
unique(rs_metadata_multi$Squirrel.ID)
# view the resampled data set
View(rs_metadata_multi)
```


```{r}
# apparently reshape2 is depricated, so this isn't ideal to use
rs_metadata_multi %>% 
  select(Squirrel.ID, Grid) %>% 
  group_by(Grid) %>% 
  reshape2::dcast(Squirrel.ID ~ Grid)
# use a tidyr alternative (not quite the same output, but can get the same information from it)
rs_reoccur <- rs_metadata_multi %>%
  pivot_wider(names_from = Grid, values_from = Squirrel.ID) %>%
  select(KL, CH, AG, SU, JO, LL)
rs_reoccur
```

Of those squirrels who were sampled multiple times, are any of them sampled in different grids?
```{r}
# check if there are values greater than zero in any of these combinations of columns
rs_counting[which(rs_counting$KL > 0 & rs_counting$AG > 0), ]
rs_counting[which(rs_counting$KL > 0 & rs_counting$CH > 0), ]
rs_counting[which(rs_counting$KL > 0 & rs_counting$JO > 0), ]
rs_counting[which(rs_counting$KL > 0 & rs_counting$LL > 0), ]
rs_counting[which(rs_counting$KL > 0 & rs_counting$SU > 0), ] # Squirrel 8815 is the only one

rs_counting[which(rs_counting$AG > 0 & rs_counting$CH > 0), ]
rs_counting[which(rs_counting$AG > 0 & rs_counting$JO > 0), ]
rs_counting[which(rs_counting$AG > 0 & rs_counting$LL > 0), ]
rs_counting[which(rs_counting$AG > 0 & rs_counting$SU > 0), ]

rs_counting[which(rs_counting$JO > 0 & rs_counting$CH > 0), ]
rs_counting[which(rs_counting$JO > 0 & rs_counting$LL > 0), ]
rs_counting[which(rs_counting$JO > 0 & rs_counting$SU > 0), ]

rs_counting[which(rs_counting$CH > 0 & rs_counting$LL > 0), ]
rs_counting[which(rs_counting$CH > 0 & rs_counting$SU > 0), ]

rs_counting[which(rs_counting$LL > 0 & rs_counting$SU > 0), ]
```

We have specific site (X, Y) information for individuals within the grid. Are there any individuals who were sampled at multiple sites within a grid?
```{r}
# Combine the x/y into a single column
rs_metadata_full$Location <- paste(rs_metadata$`Location X`, rs_metadata$`Location Y`, sep = ", ")
# Cast the location data (counts into a table)
rs_location <- rs_metadata_full %>% group_by(Squirrel.ID) %>% reshape2::dcast(Squirrel.ID ~ Location)
# Tally the rows (Total number sampled)
rs_location$Total <- rowSums(rs_location[, -1])
# Tally the number of unique columns sampled
rs_location$Different <- rowSums(rs_location[, -c(1, 379)] > 0)
# Reorder the columns so the tallies are at the front
rs_location <- rs_location %>% select(Squirrel.ID, Total, Different, everything())
# writing this second version because QIIME was having issues with the squirrel.ID name and needs the locations combined
rs_metadata_full %>% plyr::rename(replace = c("Squirrel.ID" = "SquirrelID")) %>%
  write.table("RS_metadata2.tsv", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
View(rs_location)
# how many individuals have been sampled at each location
# need to take into account grid
colSums(rs_location[, -c(1:3)] > 0) %>% as.data.frame
```
Karl's solution to above
```{r}
rs_metadata_matches %>% split(rs_metadata_matches$Squirrel.ID) %>% 
  lapply(function(x){
    c(nrow(x), nrow(distinct(x[,3:4])))
  })
```

```{r}
# where there are fewer unique sampling locations than total number of samples taken
rs_sites <- rs_location[which(rs_location$Total != rs_location$Different),]$Squirrel.ID
# where the number of unique sampling locations equals the total number of samples taken
rs_location[which(rs_location$Total == rs_location$Different),]$Squirrel.ID
```
```{r}
# is there any relationship between the number of samples collected and the number of different sites that an individual was sampled at?
ggplot(rs_location, aes(x=Total, y=Different)) + 
  geom_point() + theme_bw() + 
  scale_y_continuous(breaks=c(0,1,2,3,4,5,6,7,8,9)) +
  scale_x_continuous(breaks=c(0,3,6,9,12,15,18,21,24,27))
```

#### Sampling for site locations
```{r}
# pull out the metadata for only those squirrels sampled in different locations
rs_sites_df <- as.data.frame(rs_sites) %>% 
  rename(Squirrel.ID = rs_sites) %>%
  inner_join(rs_metadata_full, by = 'Squirrel.ID') 
View(rs_sites_df)
```
```{r}
# Write the fasta file for those squirrels sampled in different locations
rs_sites_df %>% inner_join(rs_fasta_df, by = 'SampleID') %>%
  mutate(fasta_name = paste(">", seq_name, sep="")) %>%
  select(fasta_name, sequence) %>%
  write.table("RS_sites.fasta", sep = "\n", row.names = FALSE, col.names = FALSE, quote = FALSE)
```



```{r}
# Pull the metadata for the squirrel sampled in multiple grids
rs_8815_subset <- rs_fasta_df[which(rs_metadata_full[which(rs_metadata_full$Squirrel.ID == 8815),]$SampleID == rs_fasta_df$SampleID),] %>% inner_join(rs_metadata, by = 'SampleID')
```
```{r}
# Write the fasta file for the squirrel sampled in multiple grids
rs_8815_subset %>% inner_join(rs_fasta_df, by = 'SampleID') %>%
  mutate(fasta_name = paste(">", seq_name, sep="")) %>%
  select(fasta_name, sequence) %>%
  write.table("RS_8815.fasta", sep = "\n", row.names = FALSE, col.names = FALSE, quote = FALSE)
```
### Prepare and explore the phyloseq object
Extract taxonomy information from the otu table
```{r}
rs_tax <- data.frame(run = rownames_to_column(rs_otu)$rowname,
                     Kingdom = word(rs_otu$taxonomy, 1, sep = ";") %>% word(2, sep = "__"),
                     Phylum = word(rs_otu$taxonomy, 2, sep = ";") %>% word(2, sep = "__"),
                     Class = word(rs_otu$taxonomy, 3, sep = ";") %>% word(2, sep = "__"),
                     Order = word(rs_otu$taxonomy, 4, sep = ";") %>% word(2, sep = "__"),
                     Family = word(rs_otu$taxonomy, 5, sep = ";") %>% word(2, sep = "__"),
                     Genus = word(rs_otu$taxonomy, 6, sep = ";") %>% word(2, sep = "__"),
                     Species = word(rs_otu$taxonomy, 7, sep = ";") %>% word(2, sep = "__")) %>% 
  mutate_all(na_if,"") %>% remove_rownames %>% column_to_rownames(var = "run") %>% as.matrix
```



Create a phylogenetic tree (with the whole data set)
```{r}
#align sequences
# this is quite big... doesn't wabt to run ... going to try working without the phylogeny
seqs <- getSequences("RS_sample.fasta")
names(seqs) <- seqs # This will propagate to the tip labels of the tree
alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA,verbose=FALSE)
```
```{r}
# The below code is creating the phylogeny with the aligned sequences.
library(phangorn)
phangAlign <- phyDat(as(alignment, "matrix"), type="DNA")
dm <- dist.ml(phangAlign)
treeNJ <- NJ(dm) 
fit <- pml(treeNJ, data=phangAlign)
fitGTR <- update(fit, k=4, inv=0.2)
fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE,
        rearrangement = "stochastic", control = pml.control(trace = 0))
detach("package:phangorn", unload=TRUE)
```


Create the phyloseq object
```{r}
# create the phyloseq object for the subset data
rs_sub_ps <- phyloseq(otu_table(rs_sub_otu, taxa_are_rows=FALSE), 
               sample_data(rs_sub_metadata %>% remove_rownames %>% column_to_rownames(var="SampleID")), 
               tax_table(rs_tax), phy_tree(rs_tree5$data)
               )
# create the phyloseq object for the whole data set
rs_ps <- phyloseq(otu_table(t(rs_otu_tax), taxa_are_rows=FALSE), 
               sample_data(rs_metadata %>% remove_rownames %>% column_to_rownames(var="SampleID")), 
               tax_table(rs_tax)
               )
# to add the phylogenetic tree, add the below commented script into the phyloseq object
# , phy_tree(fitGTR$tree)
```

How many of each taxonomic group are present
```{r}
table(tax_table(rs_ps)[, "Kingdom"], exclude = NULL)
table(tax_table(rs_ps)[, "Order"], exclude = NULL)
table(tax_table(rs_ps)[, "Phylum"], exclude = NULL)
table(tax_table(rs_ps)[, "Family"], exclude = NULL)
table(tax_table(rs_ps)[, "Genus"], exclude = NULL)
table(tax_table(rs_ps)[, "Species"], exclude = NULL)
```


### Plotting ideas
https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html#shiny-phyloseq
```{r}
ggplot(data = mphyseq, mapping = aes_string(x = "sex",y = "Abundance",
                              color = Color, fill = Color)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.3,
               position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    theme(legend.position="none")
```

### Qiime2 to phyloseq

```{r}
# qiime2R is from github (only qiimer is on CRAN)
#devtools::install_github("jbisanz/qiime2R")
library(qiime2R)
```

#### Based on the tutorial
https://forum.qiime2.org/t/tutorial-integrating-qiime2-and-r-for-data-visualization-and-analysis-using-qiime2r/4121


```{r}
rs_q2_metadata<-read_tsv("~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/exploration/RS_metadata_sub2.tsv")
rs_q2_metadata
```

```{r}
rs_q2_SVs<-read_qza("~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/rs-QIIME2-AH/sampled5/table.qza")
names(rs_q2_SVs) #information about the object
rs_q2_SVs$uuid
rs_q2_SVs$data
```

```{r}
rs_q2_taxonomy<-read_qza("~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/rs-QIIME2-AH/sampled5/GG-taxonomy.qza")
rs_q2_taxonomy$uuid
rs_q2_taxtable<-rs_q2_taxonomy$data %>% 
  as.tibble() %>% 
  separate(Taxon, sep="; ", c("Kingdom","Phylum","Class","Order","Family","Genus","Species"))#convert the table into a tabular split version
rs_q2_taxtable
```

```{r}
rs_q2_tree<-read_qza("~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/rs-QIIME2-AH/sampled5/rooted_tree.qza")
rs_q2_tree$uuid
rs_q2_tree$data
```

```{r}
rs_q2_shannon<-read_qza("~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/rs-QIIME2-AH/sampled5/core-metrics-phylogenetic5/shannon_vector.qza")
rs_q2_shannon$uuid
head(rs_q2_shannon$data)
```

```{r}
rs_q2_pco<-read_qza("~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/rs-QIIME2-AH/sampled5/core-metrics-phylogenetic5/unweighted_unifrac_pcoa_results.qza")
rs_q2_pco$uuid
head(rs_q2_pco$data$ProportionExplained) #this returns the variance explained
rs_q2_pco$data$Vectors[1:5, 1:3] #this returns the coordinates, here we look at the first 5 samples and 2 axes.
```

Plotting alpha diversity
```{r, message=FALSE}
# The lines connect samples from the same individual that were collected in the same location.
q2_shannon$data %>%
  as.data.frame() %>%
  rownames_to_column("SampleID") %>%
  left_join(rs_q2_metadata) %>%
  subset(SquirrelID %in% c("8745", "10275", "10342", "10374", "11176")) %>%
  mutate(Month=as.numeric(Month)) %>% #coerce this to be stored as number
    ggplot(aes(x=Month, y=shannon, group=Location, shape = Season, colour = factor(Year))) +
    geom_point(size = 2) +
    geom_line() +
    facet_grid(~SquirrelID) + #plot body sites across rows and subjects across columns
    theme_bw() +
    xlab("Month") +
    ylab("Shannon Diversity") +
    ggtitle("Shannon diversity by month")
```

Plotting beta diversity
```{r}
rs_q2_pco$data$Vectors %>%
  left_join(rs_q2_metadata) %>%
  left_join(rs_q2_shannon$data %>% rownames_to_column("SampleID")) %>%
  ggplot(aes(x=PC1, y=PC2, shape=Grid, color=Season, size=shannon)) +
  geom_point() +
  xlab(paste("PC1: ", round(100*rs_q2_pco$data$ProportionExplained[1]), "%")) +
  ylab(paste("PC2: ", round(100*rs_q2_pco$data$ProportionExplained[2]), "%")) +
  theme_bw() +
  ggtitle("Unweighted UniFrac")
```


```{r}
# so I need the OTU table to create the phyloseq object directly
rs_5qiime <- qza_to_phyloseq(features = "~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/rs-QIIME2-AH/sampled5/table.qza", 
                             tree = "~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/rs-QIIME2-AH/sampled5/rooted_tree.qza", 
                             taxonomy = "~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/rs-QIIME2-AH/sampled5/GG-taxonomy.qza",
                             metadata = "~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/exploration/RS_metadata_sub2.tsv")
#qza_to_phyloseq(tree = "../rs-QIIME2-AH/sampled5/rooted_tree.qza", metadata = "../rs-QIIME2-AH/sampled5/RS_metadata.tsv")
```



#### Some new plots with the phyloseq object

```{r}
#get the phyloseq data into a data frame
rs_5qiime_df <- psmelt(rs_5qiime)
View(rs_5qiime_df)
```

```{r}
# PCoA Ordination
Cairo::CairoJPEG(width = 1024, height = 768, file="sampled_PCoA_ordi.jpeg", bg = "transparent", quality = 75)
plot_ordination(rs_5qiime, ordinate(rs_5qiime, method = "PCoA"),
                color = sample_variables(rs_5qiime)) + 
  #ggtitle("PCoA Ordination (Bray-Curtis)") +
  stat_ellipse(type = "norm") + # not enough data to put on an elipse for all the grids
  theme_bw()
dev.off()
```

```{r}
# NMDS Ordination
plot_ordination(rs_5qiime, ordinate(rs_5qiime, method = "NMDS"),
                color = sample_variables(rs_5qiime)) + 
  ggtitle("NMDS Ordination (Bray-Curtis)") +
  stat_ellipse(type = "norm")
```

```{r}
# DCA Ordination
plot_ordination(rs_5qiime, ordinate(rs_5qiime),
                color = sample_variables(rs_5qiime)) + 
  ggtitle("DCA Ordination (Bray-Curtis)") +
  stat_ellipse(type = "norm")
```

```{r}
# CCA Ordination
plot_ordination(rs_5qiime, ordinate(rs_5qiime, method = "CCA"),
                color = sample_variables(rs_5qiime)) + 
  ggtitle("CCA Ordination (Bray-Curtis)") +
  stat_ellipse(type = "norm")
```

```{r}
# PCoA heat map
plot_heatmap(rs_5qiime, method = "PCoA", distance = "bray", 
             sample.label = "Location") + 
  ggtitle("PCoA Heat Map (5%)")

heatmap(otu_table(rs_5qiime))
```

Boxplot for Shannon Diversity Index
```{r}
rs_q2_shannon$data %>% 
  rownames_to_column("SampleID") %>%
  left_join(rs_q2_metadata) %>%
  subset(SquirrelID %in% c('10050', '10061', '10082', '10086', '10112', '10119', '10121', '10125', '10126', '10130', '10131', '10133', '10137', '10138', '10140', '10145', '10153', '10161', '10165', '10168', '10176', '10177', '10181', '10182', '10183', '10184', '10200', '10212', '10213', '10228', '10239', '10260', '10262', '10264', '10265', '10273', '10274', '10275', '10278', '10285', '10304', '10305', '10307', '10318', '10320', '10322', '10324', '10334', '10338', '10340', '10341', '10342', '10355', '10369', '10370', '10374', '10375', '10376', '10377', '10389', '10390', '10391', '10393', '10395', '10408', '10410', '10411', '10416', '10461', '10517', '10569', '10571', '10588', '10627', '10636', '10640', '10642', '10645', '10646', '10658', '10678', '10679', '10686', '10699', '10701', '10713', '10724', '10725', '10736', '10746', '10747', '10748', '10752', '10754', '10759', '10760', '10761', '10782', '10784', '10818', '10828', '10939', '10960', '11150', '11164', '11171', '11176', '11177', '11181', '11201', '11202', '11203', '11210', '11211', '11254', '11256', '11266', '11272', '11280', '11282', '11319', '11324', '11344', '11488', '11489', '11882', '13659', '13660', '13663', '13680', '13682', '13688', '13692', '8036', '8082', '8095', '8297', '8349', '8385', '8386', '8404', '8407', '8497', '8505', '8510', '8513', '8529', '8557', '8565', '8572', '8576', '8579', '8582', '8588', '8592', '8619', '8636', '8643', '8656', '8657', '8659', '8662', '8680', '8690', '8691', '8700', '8703', '8720', '8726', '8727', '8728', '8729', '8730', '8732', '8735', '8745', '8755', '8771', '8815')) %>%
  ggplot(aes(x = factor(SquirrelID), y = shannon)) +
  geom_boxplot() + theme_bw() + 
  ggtitle("Shannon Diversity of Resampled Squirrels") +
  labs(x = "Squirrel ID", y = "Shannon Diversity Index")
```

My goal here: relabel the qgraph with Squirrel ID, compensating for the fact that some squirrels were sampled more than once (need to take the average distance to that squirrel rather than just the individual samples) without using the reshape2 package
```{r}
# jaccard distance matrix
rs_jaccard <- qiime2R::read_qza("~/OneDrive - University of Guelph/Alicia's Thesis/red-squirrel-data/rs-QIIME2-AH/sampled5/core-metrics-phylogenetic5/jaccard_distance_matrix.qza")
```


```{r}
qgraph::qgraph(rs_jaccard$data, layout='spring', vsize=3)
```











